### DISCUSSION TITLE
How to authenticate to an external service with http basic auth?

### DISCUSSION THREAD

**Discussion by DenuxPlays** [goauthentik/authentik#8668] [Q&A] [answered]
So basically I have an external service that uses http basic auth with one set of credentials.
Now my goal is so that when I click on this application in authentik then should open my external service and set the http header so that I do not get the login prompt.
How do I do this or is this impossible?

What I already tried:
I tried using a proxy provider and I enabled the option to use basic http auth but sadly that didn't worked.

**Comment by Ramblurr** [comment]
I also have this question.


Referring to these docs
https://goauthentik.io/docs/providers/proxy/header_authentication
..there are two sides to it. The checkbox under the Proxy Provider about basic auth is about sending basic auth. You can configure the sending of the basic auth creds to the app with the additionalHeaders config

However to receive basic auth one has to use an app password, but as far as I can tell app passwords and proxy providers don't work. I think app passwords only work for oauth providers?

**Reply by DenuxPlays** [reply]
I read through the documentation but it couldn't help me.
The http basic auth is supplied by an caddy web server.

**Reply by moutasem1989** [reply]
http basic auth needs user:pass encoded in base64 and passed as following header:
```
proxy_set_header Authorization "Basic abcxyz”;
proxy_pass_header Authorization;
```
The http basic authentication of Authentik needs the following
```
auth_request_set $authentik_auth $upstream_http_authorization;
proxy_set_header Authorization $authentik_auth;
```

**Reply by yacob841** [reply]
> http basic auth needs user:pass encoded in base64 and passed as following header:
> 
> ```
> proxy_set_header Authorization "Basic abcxyz”;
> proxy_pass_header Authorization;
> ```
> 
> The http basic authentication of Authentik needs the following
> 
> ```
> auth_request_set $authentik_auth $upstream_http_authorization;
> proxy_set_header Authorization $authentik_auth;
> ```

I'm doing this but it's trying to send my email as the username even though I have "admin" in the username field. Any ideas?

**Reply by moutasem1989** [reply, marked as answer]
Can you try the following:
In the Nginx Proxy Manager configurations uncomment those two lines and maybe add the third:
```
auth_request_set $authentik_auth $upstream_http_authorization;
proxy_set_header Authorization $authentik_auth;
proxy_pass_header Authorization;
```
In Authentik > Admin Panel > Directory > Users or Groups: edit the desired user or group and add the following to the Attribute field
```
serviceusername: xyz
servicepassword: ***
```
In Authentik > Admin Panel >  Applications > Providers: edit the desired proxy provider. Enable `Send HTTP-Basic Authentication` and add the attributes we just created in the User/Group `serviceusername` and `servicepassword`.

Try in congnito mode or clear browser cash / history before trying.
I home this helps
